### ISSUE TITLE
Render attention maps

### ISSUE THREAD

**Issue by theVmagnificient** [jfhealthcare/Chexpert#10]
Greetings! Thanks for nice work. Is it possible to render attention map as a heat map on the picture?

**Comment by yil8** [comment]
We can add our internal heatmap.py here. Maybe we should just open source our **secret weapon** `PROB_POOL` here? @yww211

**Comment by theVmagnificient** [comment]
I would really appreciate that, thanks in advance!

**Comment by yww211** [comment]
> @theVmagnificient We can add our internal heatmap.py here. Maybe we should just open source our secret weapon `PROB_POOL` here? @yww211

Ok.

**Comment by theVmagnificient** [comment]
Hi! I've trained new model with `"global_pool": "EXP"` for 5 epochs with example.json config. But auc on test is quite low, like 0.5 mean. For how many epochs do you guys suggest to train the network?

**Comment by theVmagnificient** [comment]
Also with this model heatmaps are working a bit strange. https://drive.google.com/open?id=17fhETCe3DT3qzrQA77TwKRfEijYztbf4

**Comment by yww211** [comment]
> Hi! I've trained new model with `"global_pool": "EXP"` for 5 epochs with example.json config. But auc on test is quite low, like 0.5 mean. For how many epochs do you guys suggest to train the network?

Hi,
      i  used this branch https://github.com/jfhealthcare/Chexpert/tree/heatmap_yww with a config : `{
 "train_csv": "/nas/public/CheXpert/CheXpert-v1.0/train.csv",
 "dev_csv": "/nas/public/CheXpert/CheXpert-v1.0/valid.csv",
 "backbone": "densenet121",
 "width": 512,
 "height": 512,
 "long_side": 512,
 "fix_ratio": true,
 "pixel_mean": 128.0,
 "pixel_std": 64.0,
 "use_pixel_std": true,
 "use_equalizeHist": true,
 "use_transforms_type": "Aug",
 "gaussian_blur": 3,
 "border_pad": "pixel_mean",
 "num_classes": [
  1,
  1,
  1,
  1,
  1
 ],
 "batch_weight": true,
 "enhance_index": [
  2,
  6
 ],
 "enhance_times": 1,
 "pos_weight": [
  1,
  1,
  1,
  1,
  1
 ],
 "train_batch_size": 24,
 "dev_batch_size": 24,
 "pretrained": true,
 "log_every": 10,
 "test_every": 100,
 "epoch": 3,
 "norm_type": "BatchNorm",
 "global_pool": "EXP",
 "fc_bn": false,
 "attention_map": "None",
 "lse_gamma": 0.5,
 "fc_drop": 0,
 "optimizer": "Adam",
 "criterion": "BCE",
 "lr": 0.0001,
 "lr_factor": 0.1,
 "lr_epochs": [
  2
 ],
 "momentum": 0.9,
 "weight_decay": 0.0,
 "best_target": "auc",
 "save_top_k": 3,
 "save_index": [
  0,
  1,
  2,
  3,
  4
 ]
}`
It was trained with `"global_pool": "EXP"` as well, and we got the mean auc `0.900`, 
![image](https://user-images.githubusercontent.com/32888325/81362447-7dd8fc80-9113-11ea-8038-66455f7a003d.png)
and the heatmap is like this:
![TIM截图20200508101032](https://user-images.githubusercontent.com/32888325/81362762-2dae6a00-9114-11ea-97ea-49899eef5641.png)

i suggest that you train your model using `config.cfg` i used to reproduce our result.

**Comment by theVmagnificient** [comment]
Are you sure, that ```"attention_map": "None"``` should be non for heatmaps?

**Comment by yww211** [comment]
> Are you sure, that `"attention_map": "None"` should be non for heatmaps?

If you trained your model with `"attention_map": "None"`, then you definitely can and should use  `"attention_map": "None"`.

**Comment by theVmagnificient** [comment]
Ok, thanks

**Comment by yww211** [comment]
Hi，
We have implemented three types of heatmaps, you can check in this [Pull Request](https://github.com/jfhealthcare/Chexpert/tree/PROB_POOL).  @theVmagnificient
